### PR TITLE
config: Support environment variables

### PR DESCRIPTION
**Pull request by zhaohuxing**
* QINGSTOR_CONFIG_PATH
* QINGSTOR_ACCESS_KEY_ID
* QINGSTOR_SECRET_ACCESS_KEY

### CODE REVIEW

File: config/contract.go [hunk 1]
```diff
@@ -45,8 +45,21 @@ log_level: warn
 // DefaultConfigFile is the filename of default config file.
 const DefaultConfigFile = "~/.qingstor/config.yaml"
 
+// EnvConfigPath is config environment variable.
+const EnvConfigPath = "QINGSTOR_CONFIG_PATH"
```
Comment (Xuanwo) [inline config/contract.go]: Use `const()`instead. 

```
const (
    ...
)
```

File: config/contract.go [hunk 2]
```diff
@@ -45,8 +45,21 @@ log_level: warn
...
+const EnvAccessKeyID = "QINGSTOR_ACCESS_KEY_ID"
+
+// EnvSecretAccessKey is config envrionment variable.
+const EnvSecretAccessKey = "QINGSTOR_SECRET_ACCESS_KEY"
+
 // GetUserConfigFilePath returns the user config file path.
 func GetUserConfigFilePath() string {
+	defaultConfigFile := os.Getenv(EnvConfigPath)
+	if defaultConfigFile != "" {
+		return defaultConfigFile
```
Comment (Xuanwo) [inline config/contract.go]: May be we should convert `~/` to home here?

File: config/config.go [hunk 3]
```diff
@@ -227,6 +228,13 @@ func (c *Config) LoadConfigFromContent(content []byte) (err error) {
 	return
 }
 
+// ReadEnvVariables will set qingstor envrionment variables.
+func (c *Config) ReadEnvVariables() (err error) {
```
Comment (Xuanwo) [inline config/config.go]: - If this functions is used only in this package, we should not export it.
- `ReadEnvVariables` is not a good name for it's content, I prefer `readCredentialFromEnv`

File: config/contract.go (hunk 2 above)
Comment (Xuanwo) [reply config/contract.go]: Maybe we should add unit tests here?

Comment (zhaohuxing) [reply config/contract.go]: > May be we should convert `~/` to home here?

The environment variable value cannot be ~/

Comment (zhaohuxing) [reply config/contract.go]: > Maybe we should add unit tests here?

OK

File: config/config.go (hunk 3 above)
Comment (zhaohuxing) [reply config/config.go]: OK

File: config/contract.go (hunk 2 above)
Comment (Xuanwo) [reply config/contract.go]: > 
> > May be we should convert `~/` to home here?
> 
> The environment variable value cannot be ~/
> 

Why？Should we need to return error if we got this input?

**Comment by Xuanwo** [comment]
Please don't force push, just append new commits to keep a log for reveiws and changes.

File: config/config.go [hunk 4]
```diff
@@ -227,6 +228,12 @@ func (c *Config) LoadConfigFromContent(content []byte) (err error) {
 	return
 }
 
+func (c *Config) readCredentialFromEnv() (err error) {
+	c.AccessKeyID = os.Getenv(EnvAccessKeyID)
```
Comment (Xuanwo) [inline config/config.go:232]: What will happen if user only set `access_key_id` without `secret_access_key`?

As proposal said, we need to return error here.

Comment (zhaohuxing) [reply config/config.go:232]: It is only set here, and the check function is called later.

Comment (Xuanwo) [reply config/config.go:232]: Got it.

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**